yt_dlp/extractor/niconico.py

```diff
@@ -383,6 +396,37 @@ def extract_video_quality(video_quality):
             }
         }
 
+    def _extract_dms_format_for_quality(self, video_id, audio_quality, video_quality):
+
+        if not audio_quality.get('isAvailable') or not video_quality.get('isAvailable'):
+            return None
+
+        format_id = '-'.join(
+            ['dms', remove_start(video_quality.get('id'), 'video-'), remove_start(audio_quality.get('id'), 'audio-'), 'hls'])
+
+        vid_qual_label = video_quality.get('label')
+
+        return {
+            'url': 'niconico_dms:%s/%s/%s' % (video_id, video_quality.get('id'), audio_quality.get('id')),
+            'format_id': format_id,
+            'format_note': join_nonempty('DMS', vid_qual_label, 'hls', delim=' '),
+            'ext': 'mp4',  # Session API are used in HTML5, which always serves mp4
+            'acodec': 'aac',
+            'vcodec': 'h264',
+            'abr': float_or_none(audio_quality.get('bitRate'), scale=1000),
+            'vbr': float_or_none(video_quality.get('bitRate'), scale=1000),
+            'asr': float_or_none(audio_quality.get('samplingRate')),
```

Comment:
My bad. All sampling rates are integers:

```suggestion
            'asr': int_or_none(audio_quality.get('samplingRate')),
```